aquadoggo/src/db/stores/query.rs

```diff
@@ -0,0 +1,724 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+//! This module offers a query API to find many p2panda documents, filtered or sorted by custom
+//! parameters. The results are batched via cursor-based pagination.
+use std::collections::HashMap;
+use std::str::FromStr;
+
+use anyhow::bail;
+use p2panda_rs::document::{DocumentId, DocumentViewFields};
+use p2panda_rs::identity::PublicKey;
+use p2panda_rs::operation::OperationValue;
+use p2panda_rs::schema::{Schema, SchemaId};
+use p2panda_rs::storage_provider::error::DocumentStorageError;
+use sqlx::any::AnyRow;
+use sqlx::{query_as, FromRow};
+
+use crate::db::models::utils::parse_document_view_field_rows;
+use crate::db::models::{DocumentViewFieldRow, QueryRow};
+use crate::db::query::{
+    Cursor, Direction, Field, Filter, FilterBy, FilterSetting, LowerBound, MetaField, Order,
+    Pagination, Select, UpperBound,
+};
+use crate::db::types::StorageDocument;
+use crate::db::SqlStore;
+use crate::graphql::types::PaginationData;
+
+#[derive(Debug, Clone)]
+pub struct DocumentCursor {
```

Comment:
It would be cool to separate the cursor type from the scalar type in the GraphQL API, sticking to the separation of concerns we already have introduced in all other places.